### ISSUE TITLE
Turn on release signatures only if `-Prelease` is set

### ISSUE THREAD

**Issue by arina-ielchiieva** [apache/incubator-iceberg#545]
To following error happens happens when Iceberg project is build using JitPack:
```
* What went wrong:
A problem was found with the configuration of task ':iceberg-api:signApachePublication'.
> No value has been specified for property 'signatory.keyId'.
```
Issue is introduced after "Update build for Apache releases" commit: https://github.com/apache/incubator-iceberg/commit/2219c86ec6dc5512b2e581f500125841b1b56226

To fix this in master, release signatures should be turned on only if `-Prelease` is set, similar to how the Apache releases repository are selected.

**Comment by jzhuge** [comment]
@arina-ielchiieva Thanks for reporting the issue. Duplicate of #557 that has been fixed.

**Comment by aokolnychyi** [comment]
@arina-ielchiieva I am going to close this one as the issue should be fixed by #557, as @jzhuge noted. Feel free to reopen if that PR didn't solve the problem.
